crates/client/src/storage.rs

```diff
@@ -24,27 +24,38 @@ pub use fs::*;
 /// Package storage data must be synchronized if shared between
 /// multiple threads and processes.
 #[async_trait]
-pub trait PackageStorage: Send + Sync {
+pub trait RegistryStorage: Send + Sync {
+    ///  Loads most recent checkpoint
+    async fn load_checkpoint(&self, registry: &str)
```

Comment:
The `RegistryStorage` is already specific to a particular registry; we should revert any changes to the existing methods from the previous `PackageStorage` trait.

We should also update the comments on the trait to something like:

```
/// Trait for registry storage implementations.
///
/// Stores information such as package/operator logs and checkpoints
/// on a per-registry basis.
///
/// Registry storage data must be synchronized if shared between
/// multiple threads and processes.
```